sky/clouds/azure.py

```diff
@@ -305,14 +383,32 @@ def make_deploy_resources_variables(
         if image_id.startswith('skypilot:'):
             image_id = service_catalog.get_image_id_from_tag(image_id,
                                                              clouds='azure')
-        # Already checked in resources.py
-        publisher, offer, sku, version = image_id.split(':')
-        image_config = {
-            'image_publisher': publisher,
-            'image_offer': offer,
-            'image_sku': sku,
-            'image_version': version,
-        }
+        # Checked basic image syntax in resources.py
+        image_config = None
+        logger.info(f'YIKADEBUG checking {image_id}')
+        if image_id.startswith('/CommunityGalleries'):
+            image = self.get_community_image(image_id, region_name)
+            if image is None:
+                # Community image doesn't exist in this region need to
+                # fallback to marketplace image. Putting fallback here
+                # instead of at image validation when creating the resource
+                # because community images are regional so we need the correct
+                # region when we check whether the image exists.
+                image_id = service_catalog.get_image_id_from_tag(
+                    _FALLBACK_IMAGE_ID, clouds='azure')
```

Comment:
We should fallback only when we are using SkyPilot's custom image by default, otherwise, it is a surprising behavior to see a cluster launched with a different image although `image_id` is specified.

Reply:
good catch